### ISSUE TITLE
self.width_mult 参数为啥在model.apply 依然是None

### ISSUE THREAD

**Issue by yangzhaojason** [taoyang1122/MutualNet#6]

**Comment by yangzhaojason** [comment]
我觉得是不是在每一个网络层里面width_mult而不是self.width_mult

**Comment by taoyang1122** [comment]
@yangzhaojason Hi, which line are you refering to?

**Comment by yangzhaojason** [comment]
class USConv2d(nn.Conv2d):
    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
                 padding=0, dilation=1, groups=1, depthwise=False, bias=True,
                 us=[True, True], ratio=[1, 1]):
        in_channels_max = in_channels
        out_channels_max = out_channels
        if us[0]:
            in_channels_max = int(make_divisible(
                in_channels
                * width_mult
                / ratio[0]) * ratio[0])
        if us[1]:
            out_channels_max = int(make_divisible(
                out_channels
                * width_mult
                / ratio[1]) * ratio[1])
        groups = in_channels_max if depthwise else 1
        super(USConv2d, self).__init__(
            in_channels_max, out_channels_max,
            kernel_size, stride=stride, padding=padding, dilation=dilation,
            groups=groups, bias=bias)
        self.depthwise = depthwise
        self.in_channels_basic = in_channels
        self.out_channels_basic = out_channels
        self.width_mult = None
        self.us = us
        self.ratio = ratio

    def forward(self, input):
        in_channels = self.in_channels_basic
        out_channels = self.out_channels_basic
        if self.us[0]:
            in_channels = int(make_divisible(
                self.in_channels_basic
                * self.width_mult
                / self.ratio[0]) * self.ratio[0])
        if self.us[1]:
            out_channels = int(make_divisible(
                self.out_channels_basic
                * self.width_mult
                / self.ratio[1]) * self.ratio[1])
        self.groups = in_channels if self.depthwise else 1
        weight = self.weight[:out_channels, :in_channels, :, :]
        if self.bias is not None:
            bias = self.bias[:out_channels]
        else:
            bias = self.bias
        y = nn.functional.conv2d(
            input, weight, bias, self.stride, self.padding,
            self.dilation, self.groups)
        if getattr(FLAGS, 'conv_averaged', False):
            y = y * (max(self.in_channels_list)/self.in_channels)
        return y

**Comment by yangzhaojason** [comment]
上面代码里面，self.width_mult 你是通过model.apply初始化的，但是很遗憾，我这边没有成功。我在forward的时候，打印了self.width_mult依然为None

**Comment by taoyang1122** [comment]
@yangzhaojason  Hi, it works normally on my side. Are you able to run the training? Since if it is None when you do the forward, it should throw out an error.

**Comment by yangzhaojason** [comment]
感谢，问题已经解决
